backend/corpora/lambdas/api/v1/collection.py

```diff
@@ -110,23 +115,43 @@ def get_collection_dataset(dataset_uuid: str):
 
 @dbconnect
 def delete_collection(collection_uuid: str, visibility: str, user: str):
-    if visibility != CollectionVisibility.PRIVATE.name:
-        # Only allowed to delete private collections
-        return "", 405
     db_session = g.db_session
-    priv_collection = Collection.get_collection(
-        db_session,
-        collection_uuid,
-        CollectionVisibility.PRIVATE.name,
-        owner=_owner_or_allowed(user),
-        include_tombstones=True,
-    )
-    if priv_collection:
-        if not priv_collection.tombstone:
-            priv_collection.delete()
-        return "", 204
+    if visibility == CollectionVisibility.PUBLIC.name:
+        pub_collection = Collection.get_collection(
+            db_session,
+            collection_uuid,
+            visibility,
+            owner=_owner_or_allowed(user),
+            include_tombstones=True,
+        )
+        priv_collection = Collection.get_collection(
+            db_session,
+            collection_uuid,
+            CollectionVisibility.PRIVATE.name,
+            owner=_owner_or_allowed(user),
+            include_tombstones=True,
+        )
+
+        if pub_collection:
+            if not pub_collection.tombstone:
+                pub_collection.tombstone_collection()
```

Comment:
Is there a needed error in the case that the collection has already been tombstoned?

Reply:
No if the collection is already tombstoned, then we return 204 to indicate to the user that the deletion was and still is successful. This was an old decision made before I left.